File: lib/screens/profile_screen.dart
import 'package:flutter/material.dart';
import 'package:melodymarket/services/storage_service.dart';
import 'package:melodymarket/models/user.dart';
import 'package:melodymarket/models/album.dart';
import 'package:melodymarket/widgets/album_card.dart';

class ProfileScreen extends StatefulWidget {
  const ProfileScreen({super.key});

  @override
  State<ProfileScreen> createState() => _ProfileScreenState();
}

class _ProfileScreenState extends State<ProfileScreen> {
  User? _currentUser;
  List<Album> _userAlbums = [];
  List<Album> _purchasedAlbums = [];
  bool _isLoading = true;

  @override
  void initState() {
    super.initState();
    _loadUserData();
  }

  Future<void> _loadUserData() async {
    final user = await StorageService.getCurrentUser();
    final allAlbums = await StorageService.getAlbums();
    final purchasedAlbums = await StorageService.getUserPurchasedAlbums();
    
    setState(() {
      _currentUser = user;
      _userAlbums = allAlbums.where((album) => album.artistId == user?.id).toList();
      _purchasedAlbums = purchasedAlbums;
      _isLoading = false;
    });
  }

  Future<void> _toggleArtistMode() async {
    if (_currentUser != null) {
      final updatedUser = _currentUser!.copyWith(isArtist: !_currentUser!.isArtist);
      await StorageService.saveUser(updatedUser);
      setState(() => _currentUser = updatedUser);
      
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(
          content: Text(
            updatedUser.isArtist 
                ? 'Artist mode enabled! You can now upload music.'
                : 'Artist mode disabled.',
          ),
          backgroundColor: Theme.of(context).colorScheme.primary,
          behavior: SnackBarBehavior.floating,
        ),
      );
    }
  }

  @override
  Widget build(BuildContext context) {
    if (_isLoading) {
      return const Scaffold(
        body: Center(child: CircularProgressIndicator()),
      );
    }

    return Scaffold(
      backgroundColor: Theme.of(context).colorScheme.background,
      body: CustomScrollView(
        slivers: [
          SliverAppBar(
            expandedHeight: 200,
            pinned: true,
            backgroundColor: Theme.of(context).colorScheme.primary,
            flexibleSpace: FlexibleSpaceBar(
              background: Container(
                decoration: BoxDecoration(
                  gradient: LinearGradient(
                    begin: Alignment.topLeft,
                    end: Alignment.bottomRight,
                    colors: [
                      Theme.of(context).colorScheme.primary,
                      Theme.of(context).colorScheme.secondary,
                    ],
                  ),
                ),
                child: SafeArea(
                  child: Padding(
                    padding: const EdgeInsets.all(20),
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.end,
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          children: [
                            CircleAvatar(
                              radius: 40,
                              backgroundColor: Colors.white.withValues(alpha: 0.2),
                              backgroundImage: _currentUser?.profileImage != null
                                  ? NetworkImage(_currentUser!.profileImage!)
                                  : null,
                              child: _currentUser?.profileImage == null
                                  ? Icon(
                                      Icons.person_rounded,
                                      size: 40,
                                      color: Colors.white,
                                    )
                                  : null,
                            ),
                            const SizedBox(width: 16),
                            Expanded(
                              child: Column(
                                crossAxisAlignment: CrossAxisAlignment.start,
                                children: [
                                  Text(
                                    _currentUser?.name ?? 'User',
                                    style: Theme.of(context).textTheme.headlineSmall?.copyWith(
                                      color: Colors.white,
                                      fontWeight: FontWeight.bold,
                                    ),
                                  ),
                                  const SizedBox(height: 4),
                                  Container(
                                    padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                                    decoration: BoxDecoration(
                                      color: Colors.white.withValues(alpha: 0.2),
                                      borderRadius: BorderRadius.circular(12),
                                    ),
                                    child: Text(
                                      _currentUser?.isArtist == true 
                                          ? '🎤 Artist' 
                                          : '🎵 Music Lover',
                                      style: Theme.of(context).textTheme.labelMedium?.copyWith(
                                        color: Colors.white,
                                        fontWeight: FontWeight.w600,
                                      ),
                                    ),
                                  ),
                                ],
                              ),
                            ),
                          ],
                        ),
                      ],
                    ),
                  ),
                ),
              ),
            ),
          ),
          SliverToBoxAdapter(
            child: Padding(
              padding: const EdgeInsets.all(20),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  // Stats Cards
                  Row(
                    children: [
                      Expanded(
                        child: _StatCard(
                          title: 'Purchased',
                          value: '${_purchasedAlbums.length}',
                          icon: Icons.library_music_rounded,
                          color: Theme.of(context).colorScheme.primary,
                        ),
                      ),
                      const SizedBox(width: 16),
                      Expanded(
                        child: _StatCard(
                          title: 'Uploaded',
                          value: '${_userAlbums.length}',
                          icon: Icons.cloud_upload_rounded,
                          color: Theme.of(context).colorScheme.secondary,
                        ),
                      ),
                    ],
                  ),
                  
                  const SizedBox(height: 24),
                  
                  // Settings Section
                  Text(
                    'Settings',
                    style: Theme.of(context).textTheme.titleLarge?.copyWith(
                      fontWeight: FontWeight.bold,
                      color: Theme.of(context).colorScheme.onSurface,
                    ),
                  ),
                  const SizedBox(height: 16),
                  
                  Card(
                    elevation: 0,
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(12),
                      side: BorderSide(
                        color: Theme.of(context).colorScheme.outline.withValues(alpha: 0.2),
                      ),
                    ),
                    child: Column(
                      children: [
                        ListTile(
                          leading: Icon(
                            _currentUser?.isArtist == true 
                                ? Icons.mic_rounded
                                : Icons.person_add_rounded,
                            color: Theme.of(context).colorScheme.primary,
                          ),
                          title: Text(
                            _currentUser?.isArtist == true 
                                ? 'Disable Artist Mode'
                                : 'Enable Artist Mode',
                          ),
                          subtitle: Text(
                            _currentUser?.isArtist == true 
                                ? 'Switch back to music listener mode'
                                : 'Start uploading and selling your music',
                          ),
                          trailing: Switch(
                            value: _currentUser?.isArtist == true,
                            onChanged: (_) => _toggleArtistMode(),
                            activeColor: Theme.of(context).colorScheme.primary,
                          ),
                          onTap: _toggleArtistMode,
                        ),
                        Divider(
                          height: 1,
                          color: Theme.of(context).colorScheme.outline.withValues(alpha: 0.2),
                        ),
                        ListTile(
                          leading: Icon(
                            Icons.audio_file_rounded,
                            color: Theme.of(context).colorScheme.secondary,
                          ),
                          title: const Text('Audio Quality'),
                          subtitle: const Text('High (320 kbps)'),
                          trailing: Icon(
                            Icons.chevron_right_rounded,
                            color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.5),
                          ),
                          onTap: () {
                            ScaffoldMessenger.of(context).showSnackBar(
                              const SnackBar(
                                content: Text('Audio quality settings coming soon!'),
                              ),
                            );
                          },
                        ),
                        Divider(
                          height: 1,
                          color: Theme.of(context).colorScheme.outline.withValues(alpha: 0.2),
                        ),
                        ListTile(
                          leading: Icon(
                            Icons.download_rounded,
                            color: Theme.of(context).colorScheme.tertiary,
                          ),
                          title: const Text('Auto Download'),
                          subtitle: const Text('Download purchased music automatically'),
                          trailing: Switch(
                            value: true,
                            onChanged: (value) {
                              ScaffoldMessenger.of(context).showSnackBar(
                                const SnackBar(
                                  content: Text('Auto download settings coming soon!'),
                                ),
                              );
                            },
                            activeColor: Theme.of(context).colorScheme.tertiary,
                          ),
                        ),
                        Divider(
                          height: 1,
                          color: Theme.of(context).colorScheme.outline.withValues(alpha: 0.2),
                        ),
                        ListTile(
                          leading: Icon(
                            Icons.notifications_rounded,
                            color: Theme.of(context).colorScheme.error,
                          ),
                          title: const Text('Notifications'),
                          subtitle: const Text('New releases and recommendations'),
                          trailing: Icon(
                            Icons.chevron_right_rounded,
                            color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.5),
                          ),
                          onTap: () {
                            ScaffoldMessenger.of(context).showSnackBar(
                              const SnackBar(
                                content: Text('Notification settings coming soon!'),
                              ),
                            );
                          },
                        ),
                        Divider(
                          height: 1,
                          color: Theme.of(context).colorScheme.outline.withValues(alpha: 0.2),
                        ),
                        ListTile(
                          leading: Icon(
                            Icons.palette_rounded,
                            color: Theme.of(context).colorScheme.outline,
                          ),
                          title: const Text('Theme'),
                          subtitle: const Text('System default'),
                          trailing: Icon(
                            Icons.chevron_right_rounded,
                            color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.5),
                          ),
                          onTap: () {
                            ScaffoldMessenger.of(context).showSnackBar(
                              const SnackBar(
                                content: Text('Theme settings coming soon!'),
                              ),
                            );
                          },
                        ),
                      ],
                    ),
                  ),
                  
                  if (_userAlbums.isNotEmpty) ...[
                    const SizedBox(height: 32),
                    Text(
                      'My Uploads',
                      style: Theme.of(context).textTheme.titleLarge?.copyWith(
                        fontWeight: FontWeight.bold,
                        color: Theme.of(context).colorScheme.onSurface,
                      ),
                    ),
                    const SizedBox(height: 16),
                    GridView.builder(
                      shrinkWrap: true,
                      physics: const NeverScrollableScrollPhysics(),
                      gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
                        crossAxisCount: 2,
                        childAspectRatio: 0.8,
                        crossAxisSpacing: 16,
                        mainAxisSpacing: 16,
                      ),
                      itemCount: _userAlbums.length,
                      itemBuilder: (context, index) {
                        final album = _userAlbums[index];
                        return AlbumCard(
                          album: album,
                          isPurchased: true,
                          isDownloaded: false,
                          showPurchaseButton: false,
                          onTap: () {
                            // Navigate to album detail or edit
                          },
                        );
                      },
                    ),
                  ],
                  
                  const SizedBox(height: 32),
                  
                  // About Section
                  Card(
                    elevation: 0,
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(12),
                      side: BorderSide(
                        color: Theme.of(context).colorScheme.outline.withValues(alpha: 0.2),
                      ),
                    ),
                    child: Padding(
                      padding: const EdgeInsets.all(20),
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text(
                            'About MelodyMarket',
                            style: Theme.of(context).textTheme.titleMedium?.copyWith(
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                          const SizedBox(height: 8),
                          Text(
                            'A marketplace where artists can share their music and fans can discover and support their favorite musicians.',
                            style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                              color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.7),
                              height: 1.5,
                            ),
                          ),
                          const SizedBox(height: 12),
                          Text(
                            'Version 1.0.0',
                            style: Theme.of(context).textTheme.bodySmall?.copyWith(
                              color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.5),
                            ),
                          ),
                        ],
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _StatCard extends StatelessWidget {
  final String title;
  final String value;
  final IconData icon;
  final Color color;

  const _StatCard({
    required this.title,
    required this.value,
    required this.icon,
    required this.color,
  });

  @override
  Widget build(BuildContext context) {
    return Card(
      elevation: 0,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(12),
        side: BorderSide(
          color: Theme.of(context).colorScheme.outline.withValues(alpha: 0.2),
        ),
      ),
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Icon(
              icon,
              color: color,
              size: 24,
            ),
            const SizedBox(height: 8),
            Text(
              value,
              style: Theme.of(context).textTheme.headlineMedium?.copyWith(
                fontWeight: FontWeight.bold,
                color: Theme.of(context).colorScheme.onSurface,
              ),
            ),
            Text(
              title,
              style: Theme.of(context).textTheme.bodyMedium?.copyWith(
                color: Theme.of(context).colorScheme.onSurface.withValues(alpha: 0.7),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: src/screens/ProfileScreen.js
import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  Switch,
  Alert,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Ionicons } from '@expo/vector-icons';
import { LinearGradient } from 'expo-linear-gradient';
import { colors } from '../theme/theme';
import { StorageService } from '../services/StorageService';

export default function ProfileScreen({ navigation }) {
  const [currentUser, setCurrentUser] = useState(null);
  const [purchasedAlbums, setPurchasedAlbums] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    loadUserData();
  }, []);

  const loadUserData = async () => {
    try {
      const user = await StorageService.getCurrentUser();
      const purchased = await StorageService.getUserPurchasedAlbums();
      setCurrentUser(user);
      setPurchasedAlbums(purchased);
    } catch (error) {
      console.error('Error loading user data:', error);
    } finally {
      setIsLoading(false);
    }
  };

  const toggleArtistMode = async () => {
    if (currentUser) {
      const updatedUser = {
        ...currentUser,
        isArtist: !currentUser.isArtist,
      };
      await StorageService.saveUser(updatedUser);
      setCurrentUser(updatedUser);
      
      Alert.alert(
        'Success',
        updatedUser.isArtist 
          ? 'Artist mode enabled! You can now upload music.'
          : 'Artist mode disabled.'
      );
    }
  };

  const StatCard = ({ title, value, icon, color }) => (
    <View style={styles.statCard}>
      <View style={[styles.statIcon, { backgroundColor: color + '20' }]}>
        <Ionicons name={icon} size={24} color={color} />
      </View>
      <Text style={styles.statValue}>{value}</Text>
      <Text style={styles.statTitle}>{title}</Text>
    </View>
  );

  const QuickActionButton = ({ icon, label, onPress }) => (
    <TouchableOpacity style={styles.quickActionButton} onPress={onPress}>
      <Ionicons name={icon} size={24} color={colors.primary} />
      <Text style={styles.quickActionLabel}>{label}</Text>
    </TouchableOpacity>
  );

  if (isLoading) {
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.loadingContainer}>
          <Text>Loading profile...</Text>
        </View>
      </SafeAreaView>
    );
  }

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView>
        {/* Header */}
        <LinearGradient
          colors={[colors.primary, colors.secondary]}
          style={styles.header}
        >
          <View style={styles.profileInfo}>
            <View style={styles.avatar}>
              <Ionicons name="person" size={40} color={colors.white} />
            </View>
            <View style={styles.userInfo}>
              <Text style={styles.userName}>{currentUser?.name || 'User'}</Text>
              <View style={styles.userBadge}>
                <Text style={styles.userBadgeText}>
                  {currentUser?.isArtist ? '🎤 Artist' : '🎵 Music Lover'}
                </Text>
              </View>
            </View>
          </View>
        </LinearGradient>

        {/* Stats */}
        <View style={styles.statsContainer}>
          <StatCard
            title="Purchased"
            value={purchasedAlbums.length.toString()}
            icon="library"
            color={colors.primary}
          />
          <StatCard
            title={currentUser?.isArtist ? 'Uploaded' : 'Downloaded'}
            value="0"
            icon={currentUser?.isArtist ? 'cloud-upload' : 'download'}
            color={colors.secondary}
          />
        </View>

        {/* Artist Quick Actions */}
        {currentUser?.isArtist && (
          <View style={styles.quickActionsContainer}>
            <View style={styles.quickActionsHeader}>
              <Ionicons name="trending-up" size={20} color={colors.primary} />
              <Text style={styles.quickActionsTitle}>Artist Dashboard</Text>
            </View>
            <View style={styles.quickActionsGrid}>
              <QuickActionButton
                icon="cloud-upload"
                label="Upload"
                onPress={() => navigation.navigate('Upload')}
              />
              <QuickActionButton
                icon="analytics"
                label="Analytics"
                onPress={() => navigation.navigate('ArtistDashboard')}
              />
              <QuickActionButton
                icon="wallet"
                label="Earnings"
                onPress={() => navigation.navigate('Withdrawal', { balance: 0 })}
              />
            </View>
          </View>
        )}

        {/* Settings */}
        <View style={styles.settingsContainer}>
          <Text style={styles.sectionTitle}>Settings</Text>
          
          <View style={styles.settingsCard}>
            <TouchableOpacity style={styles.settingItem} onPress={toggleArtistMode}>
              <View style={styles.settingLeft}>
                <Ionicons
                  name={currentUser?.isArtist ? 'mic' : 'person-add'}
                  size={24}
                  color={colors.primary}
                />
                <View style={styles.settingText}>
                  <Text style={styles.settingTitle}>
                    {currentUser?.isArtist ? 'Disable Artist Mode' : 'Enable Artist Mode'}
                  </Text>
                  <Text style={styles.settingSubtitle}>
                    {currentUser?.isArtist 
                      ? 'Switch back to music listener mode'
                      : 'Start uploading and selling your music'
                    }
                  </Text>
                </View>
              </View>
              <Switch
                value={currentUser?.isArtist || false}
                onValueChange={toggleArtistMode}
                trackColor={{ false: colors.border, true: colors.primary }}
                thumbColor={colors.white}
              />
            </TouchableOpacity>

            <View style={styles.settingDivider} />

            <TouchableOpacity style={styles.settingItem}>
              <View style={styles.settingLeft}>
                <Ionicons name="musical-note" size={24} color={colors.secondary} />
                <View style={styles.settingText}>
                  <Text style={styles.settingTitle}>Audio Quality</Text>
                  <Text style={styles.settingSubtitle}>High (320 kbps)</Text>
                </View>
              </View>
              <Ionicons name="chevron-forward" size={20} color={colors.textSecondary} />
            </TouchableOpacity>

            <View style={styles.settingDivider} />

            <TouchableOpacity style={styles.settingItem}>
              <View style={styles.settingLeft}>
                <Ionicons name="download" size={24} color={colors.tertiary} />
                <View style={styles.settingText}>
                  <Text style={styles.settingTitle}>Auto Download</Text>
                  <Text style={styles.settingSubtitle}>Download purchased music automatically</Text>
                </View>
              </View>
              <Switch
                value={true}
                trackColor={{ false: colors.border, true: colors.tertiary }}
                thumbColor={colors.white}
              />
            </TouchableOpacity>

            <View style={styles.settingDivider} />

            <TouchableOpacity style={styles.settingItem}>
              <View style={styles.settingLeft}>
                <Ionicons name="notifications" size={24} color={colors.warning} />
                <View style={styles.settingText}>
                  <Text style={styles.settingTitle}>Notifications</Text>
                  <Text style={styles.settingSubtitle}>New releases and recommendations</Text>
                </View>
              </View>
              <Ionicons name="chevron-forward" size={20} color={colors.textSecondary} />
            </TouchableOpacity>
          </View>
        </View>

        {/* About */}
        <View style={styles.aboutContainer}>
          <View style={styles.aboutCard}>
            <Text style={styles.aboutTitle}>About MelodyMarket</Text>
            <Text style={styles.aboutText}>
              A revolutionary music marketplace where artists can share their music and fans can discover and support their favorite musicians. Built with love for the music community.
            </Text>
            <Text style={styles.versionText}>Version 1.0.0</Text>
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
  },
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    paddingHorizontal: 20,
    paddingVertical: 30,
  },
  profileInfo: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  avatar: {
    width: 80,
    height: 80,
    borderRadius: 40,
    backgroundColor: 'rgba(255, 255, 255, 0.2)',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 16,
  },
  userInfo: {
    flex: 1,
  },
  userName: {
    fontSize: 24,
    fontWeight: 'bold',
    color: colors.white,
    marginBottom: 8,
  },
  userBadge: {
    backgroundColor: 'rgba(255, 255, 255, 0.2)',
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 12,
    alignSelf: 'flex-start',
  },
  userBadgeText: {
    color: colors.white,
    fontSize: 14,
    fontWeight: '600',
  },
  statsContainer: {
    flexDirection: 'row',
    paddingHorizontal: 20,
    paddingVertical: 20,
    gap: 16,
  },
  statCard: {
    flex: 1,
    backgroundColor: colors.white,
    padding: 16,
    borderRadius: 12,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  statIcon: {
    width: 48,
    height: 48,
    borderRadius: 24,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 8,
  },
  statValue: {
    fontSize: 20,
    fontWeight: 'bold',
    color: colors.text,
    marginBottom: 4,
  },
  statTitle: {
    fontSize: 12,
    color: colors.textSecondary,
  },
  quickActionsContainer: {
    marginHorizontal: 20,
    marginBottom: 24,
    padding: 16,
    backgroundColor: colors.surface,
    borderRadius: 16,
    borderWidth: 1,
    borderColor: colors.primary + '20',
  },
  quickActionsHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 12,
  },
  quickActionsTitle: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.primary,
    marginLeft: 8,
  },
  quickActionsGrid: {
    flexDirection: 'row',
    gap: 12,
  },
  quickActionButton: {
    flex: 1,
    alignItems: 'center',
    padding: 12,
    backgroundColor: colors.white,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: colors.border,
  },
  quickActionLabel: {
    fontSize: 12,
    fontWeight: '600',
    color: colors.text,
    marginTop: 4,
    textAlign: 'center',
  },
  settingsContainer: {
    paddingHorizontal: 20,
    marginBottom: 24,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: colors.text,
    marginBottom: 16,
  },
  settingsCard: {
    backgroundColor: colors.white,
    borderRadius: 12,
    overflow: 'hidden',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  settingItem: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 16,
    paddingVertical: 16,
  },
  settingLeft: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
  },
  settingText: {
    marginLeft: 12,
    flex: 1,
  },
  settingTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: colors.text,
  },
  settingSubtitle: {
    fontSize: 14,
    color: colors.textSecondary,
    marginTop: 2,
  },
  settingDivider: {
    height: 1,
    backgroundColor: colors.border,
    marginLeft: 52,
  },
  aboutContainer: {
    paddingHorizontal: 20,
    paddingBottom: 40,
  },
  aboutCard: {
    backgroundColor: colors.white,
    padding: 20,
    borderRadius: 12,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  aboutTitle: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.text,
    marginBottom: 8,
  },
  aboutText: {
    fontSize: 14,
    color: colors.textSecondary,
    lineHeight: 20,
    marginBottom: 12,
  },
  versionText: {
    fontSize: 12,
    color: colors.textSecondary,
  },
});